datatypes/graph.py

```diff
@@ -0,0 +1,17 @@
+"""Defines custom data classes"""
+
+from dataclasses import dataclass
+import numpy as np
+
+
+@dataclass
+class EdgeSet:
```

Comment:
Add a one-liner describing this data class and its features. 

What is the sender and receiver? Does the graph contains edges with directions?